src/ec/eddsa.rs

```diff
@@ -112,9 +113,52 @@ impl<'a> Ed25519KeyPair {
     /// Returns the signature of the message `msg`.
     pub fn sign(&self, msg: &[u8]) -> signature::Signature {
         let mut signature_bytes = [0u8; 64];
+        let mut az = [0u8; digest::SHA512_OUTPUT_LEN];
+        digest::GFp_SHA512_4(az.as_mut_ptr(), az.len(),
+                             self.private_public.as_ptr(), 32,
+                             core::ptr::null(), 0,
+                             core::ptr::null(), 0,
+                             core::ptr::null(), 0);
         unsafe {
-            GFp_ed25519_sign(signature_bytes.as_mut_ptr(), msg.as_ptr(),
-                             msg.len(), self.private_public.as_ptr());
+            GFp_ed25519_scalar_mask(
+                slice_as_array_ref_mut!(&mut az[..SCALAR_LEN], SCALAR_LEN).unwrap())
+        };
+        let mut nonce = [0u8; digest::SHA512_OUTPUT_LEN];
+        let az_offset_ptr = unsafe { az.as_ptr().offset(32) };
```

Comment:
Done. Didn't end up using `split_at_mut` because it complicated borrowing unnecessarily in this case.